src/fstab-generator/fstab-generator.c

```diff
@@ -660,6 +661,16 @@ static int add_mount(
                 }
         }
 
+        if (flags & MOUNT_QUOTA) {
+                r = generator_hook_up_quotacheck(dest, what, where, target_unit, fstype);
```

Comment:
If a user wrongly specifies `quota` or friends to e.g. `auto` fstype, then system may fail to boot.
I think it is better to ignore the error `-EOPNOTSUPP`.
```c
r = generator_hook_up_quotacheck(...);
if (r < 0) {
        if (r != -EOPNOTSUPP)
                return r;
} else {
        r = generator_hook_up_quotaon(...);
        if (r < 0)
                return r;
}
```